### DISCUSSION TITLE
SX1262 with RF Switch does not need to be called?

### DISCUSSION THREAD

**Discussion by richonguzman** [jgromes/RadioLib#1336] [General]
(didn't know where to ask this as this is no issue:) 

have two boards here (with RP2040 and esp32s3) and both use a sx1262 version that has 1 single RF Switch for controlling Rx and Tx. So I asummed I should use something like:

radio.setRfSwitchPins(RADIO_RxEn_Pin, RADIO_TxEn_Pin); 

and it it worked as: radio.setRfSwitchPins(RADIO_RF_SWITCH, RADIO_RF_SWITCH);

then I tested: radio.setDio2AsRfSwitch();  and it worked too

the I did not use any of the two before and it worked also fine in the Rx and Tx of the sx1262 boards

so, is this really needed? or is the RadioLib library taking care of it?

boards: Waveshare RP2040-LoRa-LF and Xiao-ESP32S3-LoRa (Wio-SX1262)

**Comment by jgromes** [comment]
This depends on your hardware design. If your switch is controlled by a single input signal, and if that signal is connected to SX126x DIO2 pin, then the SX126x will control that RF switch for you. This is also the default behavior without explicitly calling `setDio2AsRfSwitch`. If your switch has two (or more) input signals, you have to call `setRfSwitchPins`.

> the I did not use any of the two before and it worked also fine in the Rx and Tx of the sx1262 boards

What is your threshold for "working fine"? If the boards are right next to each other, then you most likely will be able to communicate with RF switch in the incorrect state, with disconnected antenna etc. - simply because the distance is very short. You have to take real measurements like RSSI when receiving or measure the output power when transmitting to be sure.
